boilerpy3/extractors.py

```diff
@@ -77,7 +78,8 @@ def read_from_file(self, filename: str) -> str:
             return text_file.read()
     
     def read_from_url(self, url: str) -> str:
-        with urllib.request.urlopen(url) as url_obj:
+        request = urllib.request.Request(url, headers=self.headers)
```

Comment:
Here you can just unpack the kwargs into the request.

```suggestion
        request = urllib.request.Request(url, **request_kwargs)
```